### PR TITLE
Document restructuring

### PR DESCRIPTION
**Pull request by hassila**
Motivation:
The guide started looking a bit sprawling and not easy to get an overview, so some housekeeping seemed nice...

Changes:
Restructured documents into docs/ subfolder
Minor modifications some language to be more positive wrt Linux capabilities to be more inclusive to Linux developers coming here
Split out README.md into subdocuments for a few sections, keeping README.md tighter for good overview at first visit
Easiest to see new layout by going directly to https://github.com/hassila/guides/tree/hassila-restructuring rather than looking at diff primarily.

Result:
A better overview of the guide ready for more content and expansion.

### CODE REVIEW

**Review by tomerd — APPROVED** [review]
seems like a good idea to me!

File: README.md [hunk 1]
```diff
@@ -1,98 +1,17 @@
...
-- If you don't have `--privileged` (or `--security-opt seccomp=unconfined`) containers (meaning you won't be able to use `lldb`) or you don't want to use lldb, consider using a library like [`swift-backtrace`](https://github.com/swift-server/swift-backtrace) to get stack traces on crash.
-
-## Deployment to Servers or Public Cloud
-
-Once an application is built for production, it needs to be packaged for deployment. There are several strategies for packaging Swift applications for deployment, see the [Packaging Guide](packaging.md) for more information.
-
-A separate guide for [Ubuntu on DigitalOcean](digital-ocean.md) is also available.
-
-TODO: add guides for popular public clouds like AWS, GCP, Azure, Heroku, etc.
+_The guide is a community effort, and all are invited to share their tips and know-how. Please provide a PR if you have ideas for improving this guide!_
```
Comment (ktoso) [inline README.md:17]: 👍 

(and huge thanks for your recent efforts here!)

**Review by ktoso — APPROVED** [review]
LGTM, seems great to me 👍

**Comment by weissi** [comment]
Fantastic, thank you so much @hassila !!

**Pull request merged**